### PR TITLE
test: FormLabel dedicated directory and tests

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
This PR only includes tests for the FormLabel component but no storybook. The FormLabel is a React Bootstrap component that we aim to deprecate asap, so the goal of this PR is only to contribute to the stability of the system as it currently stands with no additional investment. 

This PR also moves the FormLabel component to a dedicated directory.

### CODE REVIEW

**Comment by geido** [comment]
@junlincc @rusackas

**Comment by junlincc** [comment]
Thanks for the PR! this project is progressing so nicely, thanks to you all @geido @michael-s-molina @yardz 🙏

image

File: superset-frontend/src/components/FormLabel/FormLabel.test.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  expect(screen.getByLabelText('Form label')).toBeInTheDocument();
+});
+
+test('should be shown as required', () => {
+  const requiredProps = {
+    ...mockedProps,
+    required: true,
+  };
+  render(<FormLabel {...requiredProps} />);
+  expect(screen.getByTestId('span-required')).toBeInTheDocument();
```
Comment (michael-s-molina) [inline superset-frontend/src/components/FormLabel/FormLabel.test.tsx:51]: I think we can replace the ID with getByText and parentNode.
`expect(screen.getByText('*').parentNode). toBeInTheDocument();`

File: superset-frontend/src/components/FormLabel/index.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ export default function FormLabel({
       <ControlLabel htmlFor={htmlFor}>
         {children}
         {required && (
-          <span className="text-danger m-l-4">
+          <span className="text-danger m-l-4" data-test-id="span-required">
```
Comment (michael-s-molina) [inline superset-frontend/src/components/FormLabel/index.tsx:39]: We can remove this ID if we access the element by `parentNode`

**Review by michael-s-molina — COMMENTED** [review]
Just a suggestion to remove the ID and use `getByText` and `parentNode`.

**Review by michael-s-molina — APPROVED** [review]

**Review by rusackas — APPROVED** [review]

**Pull request merged**